src/solvers/dg_unstructured_quad/dg_2d.jl

```diff
@@ -235,15 +235,22 @@ function calc_boundary_flux!(cache, t, boundary_conditions,
     boundary_condition = boundary_conditions[ name[boundary] ]
 
     # calc boundary flux on the current boundary interface
-    for j in eachnode(dg)
-      calc_boundary_flux!(surface_flux_values, t, boundary_condition, mesh, equations, dg, cache,
-                          j, side, element, boundary)
-    end
+    calc_boundary_flux!(surface_flux_values, t, boundary_condition, mesh, equations, dg, cache,
+                        side, element, boundary)
   end
 
   return nothing
 end
 
+# use a function barrier for now to improve type stability
+@noinline function calc_boundary_flux!(surface_flux_values, t, boundary_condition::BC,
+                                       mesh::UnstructuredQuadMesh, equations, dg::DG, cache,
+                                       side, element, boundary) where {BC}
```

Comment:
Just a point of clarification. Defining `boundary_condition::BC` and then having `where {BC}` is helping Julia keep track on the type now that the `boundary_condition` is a `Symbol` instead of a `String`, right?

Reply:
Julia's compiler uses some heuristics to decide whether or not to specialize a method on the type of each argument, see https://docs.julialang.org/en/v1/devdocs/functions/#compiler-efficiency-issues. This ensures that the method will be specialized on the type of `boundary_condition`, which is a function. It doesn't have to do anything with `Symbol` vs. `String`.